### PR TITLE
Add vendoring test

### PR DESCRIPTION
**Pull request by bkcsoft**
> To build using the main module's top-level vendor directory to satisfy
dependencies (disabling use of the usual network sources and local
caches), use 'go build -mod=vendor'.

Slightly annoying...

> The -mod flag was added with Go 1.11 and therefore is not available in the older version of Go.  

This means we'd drop support for Go <1.11 :unamused: unless we do some `Makefile` magic :thinking: 

reference: https://arslan.io/2018/08/26/using-go-modules-with-vendor-support-on-travis-ci/

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -28,11 +30,15 @@ ui/lib/src/api: swagger.yaml
 
 .PHONY: lint
 lint:
-	golint $(shell go list ./pkg/gitpods/...)
+	$(GO) mod tidy
+	$(GO) mod vendor
+	git update-index --refresh
+	git diff-index --quiet HEAD
```
Comment (bkcsoft) [inline Makefile]: `update-index --refresh` so that touched-but-unchanged don't mess up `diff-index` :roll_eyes:

Comment (bkcsoft) [reply Makefile]: Also, those are plumbing which means they have proper exit-codes that `make` understands... (hence not `diff --quiet`)

Comment (metalmatze) [reply Makefile]: As these new commands do a lot more, can we put them in a different make target? In CI I'm fine with running both make targets in the same lint step though.

**Comment by metalmatze** [comment]
These changes are to make sure that `vendor/` is up-to-date with what the `go.sum` says, right?
Not a fan of running this in every single CI pipeline, but if we want to be as strict about it we should probably do that.
In other projects I mostly made sure that updates to `Gopkg.toml` got reflected in `vendor/`. Not sure if that's different with Go modules.

Yes, we only support Go 1.11+. I think that is fine for now. But we should probably stick to supporting the last 2 versions of Go after that.

**Comment by metalmatze** [comment]
Can you update this PR according to the comments. Otherwise I'll simply push to your branch and merge it in the next few hours. :blush:

**Comment by metalmatze** [comment]
We'll see if it might become a problem. Let's see. It's not perfect, but enough for now I think.

**Pull request merged**